packages/blockchain-wallet-v4-frontend/src/modals/BuySell/UpdateSecurityCode/UpdateSecurityCode.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { useDispatch, useSelector } from 'react-redux'
+import {
+  Button,
+  Flex,
+  IconCheckCircle,
+  IconCreditCard,
+  IconLockClosed,
+  IconShield,
+  Input,
+  Logo,
+  Padding,
+  PaletteColors,
+  SemanticColors,
+  SpinningLoader,
+  Text
+} from '@blockchain-com/constellation'
+
+import { Card } from 'components/Card'
+import {
+  FlyoutContainer,
+  FlyoutContent,
+  FlyoutFooter,
+  FlyoutHeader
+} from 'components/Flyout/Layout'
+import { StandardRow } from 'components/Rows'
+import { actions, selectors } from 'data'
+import { useRemote } from 'hooks'
+
+import { UpdateSecurityCodeComponent } from './types'
+
+const UpdateSecurityCode: UpdateSecurityCodeComponent = ({ backToEnterAmount }) => {
+  const dispatch = useDispatch()
+  const [cvv, setCvv] = useState<string | null>(null)
+  const { data: order } = useRemote(selectors.components.buySell.getBSOrder)
+  const {
+    hasData: cvvHasData,
+    hasError: cvvHasError,
+    isLoading: cvvLoading
+  } = useRemote(selectors.components.buySell.getCvvStatus)
+  const method = useSelector(selectors.components.buySell.getBSPaymentMethod)
+
+  const updateCvv = () => {
+    const paymentId = order?.attributes?.paymentId
+    if (cvv && paymentId) {
+      dispatch(actions.components.buySell.updateCardCvv({ cvv, paymentId }))
+    }
+  }
+
+  useEffect(() => {
+    if (cvvHasData) {
+      setTimeout(() => {
```

Comment:
Do we absolutely need the timeout here and how reliable is it?

Reply:
I put this slight delay in order to show the user the green check in the postfix of the input as shown in the screenshot. Otherwise the once they submit the screen will just go to the next step with no confirmation to the user.
![image](https://user-images.githubusercontent.com/57680122/205168178-c44e20d8-8099-4e7c-8f05-beb478f1e7f0.png)

Reply:
Makes sense.

By the way, even though it is a slight delay, you might want to clear the timeout in the cleanup function to avoid unexpected behaviour in edge cases.